boards/atsame54_xpro/memory.x

```diff
@@ -0,0 +1,33 @@
+MEMORY
+{
+  FLASH (rx) : ORIGIN = 0x00000000+0x4000, LENGTH = 0x100000-0x4000 /* First 16KB used by bootloader */
```

Comment:
To answer your question about no bootloader, please remove the +0x4000 and -0x4000 here.

Reply:
Yeah, it's probably best not to expect a bootloader if there is none by default.